### PR TITLE
CARRY: Add workflow to release ODH/CFO with compiled test binaries

### PR DESCRIPTION
**Pull request by sutaakar**
## Description

As part of reducing downstream test execution time effort (https://issues.redhat.com/browse/RHOAIENG-3757) I would like to compile test binaries for every release and store them in GitHub release. These compiled binaries can be used for downstream testing, significantly reducing test time.

## How Has This Been Tested?

Manually.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by sutaakar** [comment]
`e2e-odh-cfo` failure is related to missing `DSCInitialization`. It will be addressed separately.

File: .github/workflows/odh-release.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Compile tests
+      run: |
+        go test -c -o compiled-tests/e2e ./test/e2e/
+        go test -c -o compiled-tests/odh ./test/odh/
+
+    - name: Creates a release in GitHub
+      run: |
+        gh release create ${{ github.event.inputs.version }} --target ${{ github.ref }} compiled-tests/*
+      env:
+        GITHUB_TOKEN: ${{ secrets.CODEFLARE_MACHINE_ACCOUNT_TOKEN }}
```
Comment (astefanutti) [inline .github/workflows/odh-release.yml:48]: Is it the right token?

Comment (sutaakar) [reply .github/workflows/odh-release.yml:48]: Yes, `codeflare-machine-account` has a maintainer role for `opendatahub-io/codeflare-operator`

**Comment by sutaakar** [comment]
Merging manually as the test failures are not related (to be addressed separately)

**Pull request merged**